src/System.Net.Requests/src/System/Net/RequestStream.cs

```diff
@@ -99,16 +99,52 @@ public override void SetLength(long value)
 
         public override void Write(byte[] buffer, int offset, int count)
         {
+            if (buffer == null)
+            {
+                throw new ArgumentNullException(nameof(buffer));
+            }
+            if (offset < 0 || offset > buffer.Length)
+            {
+                throw new ArgumentOutOfRangeException(nameof(offset));
+            }
+            if (count < 0 || count > (buffer.Length - offset))
+            {
+                throw new ArgumentOutOfRangeException(nameof(count));
+            }
             _buffer.Write(buffer, offset, count);
         }
 
         public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
         {
+            if (buffer == null)
+            {
+                throw new ArgumentNullException(nameof(buffer));
+            }
+            if (offset < 0 || offset > buffer.Length)
+            {
+                throw new ArgumentOutOfRangeException(nameof(offset));
+            }
+            if (count < 0 || count > (buffer.Length - offset))
+            {
+                throw new ArgumentOutOfRangeException(nameof(count));
```

Comment:
Can you try running these on desktop? (Make sure you've done a build with `build.cmd -framework=netfx` and then from the tests directory do `msbuild /t:rebuildandtest /p:TargetGroup=netfx`. I expect some of these are going to fail, e.g. complaining about this being named `size` instead of `count`.  That's because of what's arguably a bug in the desktop implementation: it uses different names for the arguments to the BeginRead/Write methods, and it doesn't override the Read/WriteAsync methods, so it picks up the base implementation which delegates from Read/WriteAsync to BeginRead/Write, thereby picking up validation code that refers to one set of parameter names even though the entry point the developer used was with another.  You can use this helper to specify the parameter name to be expected in core and the one to be expected in desktop:
https://github.com/dotnet/corefx/blob/master/src/Common/tests/System/AssertExtensions.cs#L22